### PR TITLE
Add RavenDB provider. #111

### LINKED ISSUE #111: Consider RavenDB support
Please consider support [Serilog.Sinks.RavenDB](https://github.com/ravendb/serilog-sinks-ravendb)

### PR DESCRIPTION
**Pull request by mo-esmp**
Hey @followynne 
I need your help. Three tests fail, and it's related to UTC datetime and I could figure out what's the problem. Could you please check when you have time?

### CODE REVIEW

**Comment by followynne** [comment]
ciao @mo-esmp 

you can find the fix in this [commit](https://github.com/followynne/serilog-ui/commit/91911747c09747ca0124f12857982935b1504d57) :) 
[didn't have permission to push it directly on this branch, sry]

**Comment by mo-esmp** [comment]
I applied your suggestion and got the same result.
- The `It_finds_data_with_all_filters_by_utc` test fails because the properties retrieved from the database are empty, but properties  of [sample logs](https://github.com/serilog-contrib/serilog-ui/blob/8eb0e8d84f1b06a695600f328b86ad211bdddc6d/tests/Serilog.Ui.Common.Tests/DataSamples/ElasticSearchLogModelFaker.cs#L47) are not empty
- The `It_finds_only_data_emitted_in_dates_range_by_utc` fails because the returned result is empty. I did manual tests and the date filter worked as expected

I got stuck and have no idea how I can fix these two tests :-/

**Comment by followynne** [comment]
Ciao @mo-esmp 

you missed the changes from tests/Serilog.Ui.RavenDbProvider.Tests/DataProvider/DataProviderSearchTest.cs in that commit, try to apply it and at least the all-filters should work :) after that update, I'll refetch your branch and check if the tests still not pass!

**Comment by mo-esmp** [comment]
Sorry, I didn't get you. What changes did I miss?

**Comment by followynne** [comment]
I meant this [file](https://github.com/followynne/serilog-ui/blob/91911747c09747ca0124f12857982935b1504d57/tests/Serilog.Ui.RavenDbProvider.Tests/DataProvider/DataProviderSearchTest.cs) but, if it's okay for you, you can skip the failing tests - in the 3.0 development wip I'm unifying and updating all tests, thus I can re-enable these ones on that branch :)

**Comment by mo-esmp** [comment]
Ciao @followynne 
Please review the PR whenever you have time.

File: src/Serilog.Ui.RavenDbProvider/Extensions/SerilogUiOptionBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    ///   Configures the SerilogUi to connect to a RavenDB database.
+    /// </summary>
+    /// <param name="optionsBuilder">The Serilog UI options builder.</param>
+    /// <param name="documentStore">A DocumentStore for a RavenDB database.</param>
+    /// <param name="collectionName"> Name of the collection to query logs. default value is <c>LogEvents</c>.</param>
+    /// <exception cref="ArgumentNullException">throw if documentStore is null</exception>
+    public static void UseRavenDb(this SerilogUiOptionsBuilder optionsBuilder, IDocumentStore documentStore, string collectionName = "LogEvents")
+    {
+        Guard.Against.Null(documentStore, nameof(documentStore));
+        Guard.Against.Null(documentStore.Urls, nameof(documentStore.Urls));
```
Comment (followynne) [inline src/Serilog.Ui.RavenDbProvider/Extensions/SerilogUiOptionBuilderExtensions.cs]:
```suggestion
```

Comment (followynne) [reply src/Serilog.Ui.RavenDbProvider/Extensions/SerilogUiOptionBuilderExtensions.cs]: NullOrEmpty on ln 24 already perform this check

File: src/Serilog.Ui.RavenDbProvider/RavenDbDataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private readonly IDocumentStore _documentStore;
+
+    public RavenDbDataProvider(IDocumentStore documentStore, string collectionName)
+    {
+        _documentStore = documentStore;
+        _collectionName = collectionName;
+    }
+
+    /// <inheritdoc/>
+    public string Name => string.Join(".", "RavenDB", "LogEvents");
```
Comment (followynne) [inline src/Serilog.Ui.RavenDbProvider/RavenDbDataProvider.cs]:
```suggestion
    public string Name => string.Join(".", "RavenDB");
```

File: src/Serilog.Ui.RavenDbProvider/Serilog.Ui.RavenDbProvider.nuspec [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		<authors>Mohsen Esmailpour</authors>
+		<owners>mo.esmp</owners>
+		<projectUrl>https://github.com/serilog-contrib/serilog-ui</projectUrl>
+		<license type="expression">MIT</license>
+		<requireLicenseAcceptance>false</requireLicenseAcceptance>
+		<description>RavenDB data provider for Serilog UI.</description>
+		<readme>docs\README.md</readme>
+		<releaseNotes></releaseNotes>
+		<copyright></copyright>
+		<tags>serilog serilog-ui serilog.sinks.postgresql</tags>
```
Comment (followynne) [inline src/Serilog.Ui.RavenDbProvider/Serilog.Ui.RavenDbProvider.nuspec]:
```suggestion
		<tags>serilog serilog-ui serilog.sinks.ravendb</tags>
```

**Review by followynne — APPROVED** [review]
Great job, LGTM! 🚀

**Review by followynne — APPROVED** [review]

**Pull request merged**